bson/src/main/org/bson/codecs/BigDecimalCodecProvider.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs;
+
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+
+/**
+ * A {@code BigDecimal} codec provider
+ *
+ * @since 3.5
+ */
+public final class BigDecimalCodecProvider implements CodecProvider {
+    private static final BigDecimalCodec BIG_DECIMAL_CODEC = new BigDecimalCodec();
```

Comment:
No need to cache in a global.  The CodecRegistry ensures that it never asks for the same Codec twice.